Site/dataobjects/SiteAttachment.php

```diff
@@ -1,556 +1,448 @@
+        $this->id_field = 'integer:id';
+
+        $this->registerDateProperty('createdate');
+
+        $this->registerInternalProperty(
+            'attachment_set',
+            SwatDBClassMap::get(SiteAttachmentSet::class)
+        );
+    }
+
+    public static function setCDNBase($base)
+    {
+        self::$cdn_base = $base;
+    }
+
+    public function getFormattedFileSize()
+    {
+        return SwatString::byteFormat($this->file_size, -1, false, 1);
+    }
+
+    public function getValidMimeTypes()
+    {
+        $ms_openxml_prefix = 'application/vnd.openxmlformats-officedocument';
+
+        return ['audio/mp4', 'video/mp4', 'audio/mpeg', 'application/zip', 'application/pdf', 'image/jpeg', 'image/png', 'text/html', 'application/msword', 'application/vnd.ms-excel', $ms_openxml_prefix . '.wordprocessingml.document', $ms_openxml_prefix . '.spreadsheetml.sheet'];
+    }
+
+    /**
+     * Returns the extension of the attachment based on  mime type.
+     *
+     * For MPEG-4 Audio, we use the non-standard but universally accepted m4a
+     * extension. See wikipedia for more details
+     * {@link https://en.wikipedia.org/wiki/.m4a}.
+     *
+     * @returns string The extension of the file.
+     *
+     * @param mixed|null $mime_type
+     */
+    public function getExtension($mime_type = null)
+    {
+        if ($mime_type == '') {
+            $mime_type = $this->mime_type;
+        }
+
+        $ms_openxml_prefix = 'application/vnd.openxmlformats-officedocument';
+
+        $map = ['audio/mp4' => 'm4a', 'video/mp4' => 'mp4', 'audio/mpeg' => 'mp3', 'application/zip' => 'zip', 'application/pdf' => 'pdf', 'image/jpeg' => 'jpg', 'image/png' => 'png', 'text/html' => 'html', 'application/msword' => 'doc', 'application/vnd.ms-excel' => 'xls', $ms_openxml_prefix . '.wordprocessingml.document' => 'docx', $ms_openxml_prefix . '.spreadsheetml.sheet' => 'xlsx'];
+
+        if (!array_key_exists($mime_type, $map)) {
+            throw new SiteException(
+                sprintf(
+                    'Unknown mime type %s',
+                    $mime_type
+                )
+            );
+        }
+
+        return $map[$mime_type];
+    }
+
+    public function getHumanFileType($mime_type = null)
+    {
+        if ($mime_type == '') {
+            $mime_type = $this->mime_type;
+        }
+
+        $ms_openxml_prefix = 'application/vnd.openxmlformats-officedocument';
+
+        $map = ['audio/mp4' => Site::_('M4A'), 'video/mp4' => Site::_('MP4'), 'audio/mpeg' => Site::_('MP3'), 'application/zip' => Site::_('ZIP'), 'application/pdf' => Site::_('PDF'), 'image/jpeg' => Site::_('JPEG Image'), 'image/png' => Site::_('PNG Image'), 'text/html' => Site::_('Web Document'), 'application/msword' => Site::_('Word Document'), 'application/vnd.ms-excel' => Site::_('Excel Document'), $ms_openxml_prefix . '.wordprocessingml.document' => Site::_('Word Document'), $ms_openxml_prefix . '.spreadsheetml.sheet' => Site::_('Excel Document')];
```

Comment:
These maps could possible be improved with match expressions.

Reply:
Refactored.